src/main/java/org/jabref/model/entry/Date.java

```diff
@@ -0,0 +1,126 @@
+package org.jabref.model.entry;
+
+import java.time.LocalDate;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalAccessor;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class Date {
+
+    private final TemporalAccessor date;
+
+    public Date(TemporalAccessor date) {
+        this.date = date;
+    }
+
+    /**
+     * Try to parse the following formats
+     *  - "M/y" (covers 9/15, 9/2015, and 09/2015)
+     *  - "MMMM (dd), yyyy" (covers September 1, 2015 and September, 2015)
+     *  - "yyyy-MM-dd" (covers 2009-1-15)
+     *  - "dd-MM-yyyy" (covers 15-1-2009)
+     *  - "d.M.uuuu" (covers 15.1.2015)
+     *  - "uuuu.M.d" (covers 2015.1.15)
+     * The code is essentially taken from http://stackoverflow.com/questions/4024544/how-to-parse-dates-in-multiple-formats-using-simpledateformat.
+     */
+    public static Optional<Date> parse(String dateString) {
```

Comment:
Theoretically we should support all allowed date formats from biblatex: EDTF format
2.3.8 Date and Time Specifications
A while ago I looked into that and found that there is no easy way to support all in java.